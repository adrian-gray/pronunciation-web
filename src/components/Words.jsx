import React from 'react'

import {
  Paper,
  Table,
  TableBody,
  TableCell,
  TableRow,
  Typography,
  withStyles
} from '@material-ui/core'

import SplitHilite from './SplitHilite'

const styles = (theme) => ({
  headspace: theme.headspace,
  largeText: theme.largeText,
  subCanvas: theme.subCanvas
})

const Words = (props) => {
  const { classes, tag, words } = props
  const list = Object.entries(words).map((arr, index) => {
    return (
      <TableRow key={index}>
        <TableCell>
          <Typography className={classes.largeText}>
            <SplitHilite str={arr[0]} />
          </Typography>
        </TableCell>
        <TableCell>
          <Typography className={classes.largeText}>
            <SplitHilite str={arr[1]} />
          </Typography>
        </TableCell>
      </TableRow>
    )
  })

  return (
    <div className={classes.headspace}>
      <Typography variant='title' gutterBottom>
        {`Words using ${tag}`}
      </Typography>

      <Paper className={classes.subCanvas}>
        <Table className={classes.root}>
          <TableBody>
            {list}
          </TableBody>
        </Table>
      </Paper>
    </div>
  )
}

export default withStyles(styles)(Words)
