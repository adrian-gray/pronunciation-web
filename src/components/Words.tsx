import React from "react";

import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";

import SplitHilite from "./SplitHilite";

export default props => {
  const { ipa, words } = props;
  const list = Object.entries(words).map((arr, index) => {
    return (
      <tr key={index}>
        <td>
          <p className="large-text">
            <SplitHilite str={arr[0]} />
          </p>
        </td>
        <td>
          <p className="large-text">
            <SplitHilite str={arr[1]} />
          </p>
        </td>
      </tr>
    );
  });

  return (
    <Container className="headspace">
      <h4>
        {`Words using `}
        <SplitHilite str={ipa} />
      </h4>

      <Container>
        <Table>
          <tbody>{list}</tbody>
        </Table>
      </Container>
    </Container>
  );
};
