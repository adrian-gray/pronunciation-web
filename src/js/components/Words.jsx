import React from 'react'

import { splitHilite } from './../Utils'

export default function PronounceText (props) {
  const { words } = props
  const list = Object.entries(words).map( (arr, i) => {
    return (
      <tr key={i}>
        <td>{splitHilite(arr[0])}</td>
        <td>{splitHilite(arr[1])}</td>
      </tr>
    )
  })

  return (
    <table className='table table-sm'>
      <thead>
        <tr>
          <th colSpan='2'>Words</th>
        </tr>
      </thead>
      <tbody>
        {list}
      </tbody>
    </table>
  )
}
