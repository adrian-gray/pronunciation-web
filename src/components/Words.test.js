import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Words from "./Words";

afterEach(cleanup);

test("Words", async () => {
  const props = {};

  const container = shallow(<Words {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
