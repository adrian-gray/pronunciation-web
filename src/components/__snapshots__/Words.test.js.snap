// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Words 1`] = `
<div>
  <div
    class="makeStyles-headspace-1"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Words using 
      <span>
        Pirate
      </span>
       
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-subCanvas-3 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  0
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  car
                </span>
                 
              </p>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  1
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  cat
                </span>
                 
              </p>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  2
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  cad
                </span>
                 
              </p>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  3
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  cap
                </span>
                 
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
